pkg/webhook/internal/metrics/metrics.go

```diff
@@ -32,9 +32,30 @@ var (
 		},
 		[]string{"webhook"},
 	)
+
+	// RequestTotal is a prometheus metric which is a counter of the total processed admission requests.
+	RequestTotal = func() *prometheus.CounterVec {
+		return prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "controller_runtime_webhook_requests_total",
+				Help: "Total number of admission requests by HTTP status code.",
+			},
+			[]string{"webhook", "code"},
+		)
+	}()
+
+	// RequestInFlight is a prometheus metric which is a gauge of the in-flight admission requests.
+	RequestInFlight = func() *prometheus.GaugeVec {
+		return prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "controller_runtime_webhook_requests_in_flight",
```

Comment:
technically a conversion hook is also a `webhook`, would it maybe make sense to add the `admission` somewhere in the name to avoid name clashes?

Reply:
I'm honestly not sure what the best name here but the `controller_runtime_webhook_` prefixed used I got from [`RequestLatency`](https://github.com/kubernetes-sigs/controller-runtime/blob/1fb86b6/pkg/webhook/internal/metrics/metrics.go#L33) since these metrics apply for the same piece of code.

It maybe better to change these names in a follow up PR to avoid a BC break within this PR?

Reply:
Yeah, makes sense